### PR TITLE
Add a very simple wrapper (in_series) around :meck.seq

### LINKED ISSUE #124: Return different values depending on number of times a function is called
Background
----

I am working in a project that uses this library for testing. However I need to run a specific scenario, where the output of a function depends on the number of times said function is called. 
I have read the documentation and I don't think it is possible to do this with Mock, but perhaps I am missing something.

Code
----

In this test I have a Storage module which I want to mock (it has side effects and is a boundary). 
In the test, I call a function `get`, which the first time returns `nil`, then I save some data with `save` and then I call get again.

```
test_with_mock "returns OK when products list is saved into storage", Storage, [],
    [
      save: fn _table, data -> {:ok, data} end,
      get: fn
        _table, _seats_number -> {:ok, nil} # first call it returns nil
        _table, _seats_number -> {:ok, [1]} # second call should return some data
      end
    ] do
      # Arrange
      products = [
        %{"id" => 1, "gems" => 4},
        %{"id" => 3, "gems" => 4},
      ]
      products_table = :products

      # Act & Assert
      actual = Engine.save_products(products)
      expected = {:ok, :list_saved_successfully}

      assert actual == expected
      
      # First call to get returns nil because the table is empty. 
      # Then we save something into it.
      assert_called Storage.get(products_table, 4)
      assert_called Storage.save(products_table, {4, [1]})

      # Second call should return the product previously saved
      # But the mock only returns nil
      assert_called Storage.get(products_table, 4)
    end
```

The issue here is that since there is no counter, I don't have a way of returning a different output depending on the number of calls a function was called.

To be fair, Mock does offer a way to returns different outputs when the _input_ is different. However, this is not the case. The input is the same, the only thing different is the number of invocations.

I am aware of call history and meck, but as far as I understand, this is used more for asserts and not used to return specific outputs (as in, this is a history stack).

Question
----

How can I achieve my goal using Mock? (is it possible?)

### PR DESCRIPTION
**Pull request by tpitale**
Possibly resolves #124

cc @Olshansk

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -237,6 +238,34 @@ end
...
+  use ExUnit.Case, async: false
+
+  import Mock
+
+  test "mock repeated calls with in_series" do
+    with_mock String,
+      [slice: [in_series(["test", 1..3], ["string1", "string2", "string3"])]]
+    do
+      assert String.slice("test", 1..3) == "string1"
+      assert String.slice("test", 1, 3) == "string2"
```
Comment (Olshansk) [inline README.md:261]: Why `1, 3` instead of `1..3`?

Comment (tpitale) [reply README.md:261]: Typo. Will fix.

File: README.md [hunk 2]
```diff
@@ -237,6 +238,35 @@ end
...
 
+## Mock repeated calls
+
+You can mock repeated calls to the same function _and_ arguments to return
+different results in a series using the `in_series` call with static values.
+This does not currently support _functions_.
+
+**Caution**: This is only useful in rare instances where the underlying business
+logic is likely to be stateful. If you can avoid it by using different function
+arguments, or refactor the function to be stateful, consider that approach first.
```
Comment (tpitale) [inline README.md:249]: Should this be `to _not_ be stateful`?

**Comment by tpitale** [comment]
@Olshansk Updated, sorry for the loooong delay.

File: README.md (hunk 2 above)
Comment (Olshansk) [reply README.md:249]: I believe it is correct as is. If the function is not stateful, it should be idempotent and deterministic.

The same input will always produce the same output, regardless of how many times it's called. If there is state or entropy, that is the only case where the expected output will be different depending on the call.

Comment (Olshansk) [reply README.md:249]: Let me know what you think. I can merge this in if you 👍 my response.

Comment (tpitale) [reply README.md:249]: Thanks!

**Pull request merged**